GAME/TOMB4FX.C

```diff
@@ -355,14 +389,162 @@ void TriggerBlood(int x, int y, int z, int a4, int num)
 
 void TriggerExplosionBubble(int x, int y, int z, short room_num)
 {
-	S_Warn("[TriggerExplosionBubble] - Unimplemented!\n");
+	int dx = lara_item->pos.x_pos - x;
+	int dz = lara_item->pos.z_pos - z;
+	if (dx >= -16384 && dx <= 16384 && dz >= -16384 && dz <= 16384)
+	{
+		struct SPARKS* sptr = &spark[GetFreeSpark()];
```

Comment:
All variables must be declared at the start of a method.

Reply:
C89 states that variables must be declared at the start of a scope block, which is the case here